src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -25,18 +36,119 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         debugConfiguration: AwsSamDebuggerConfiguration,
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
-        if (!AWS_SAM_DEBUG_REQUEST_TYPES.has(debugConfiguration.request)) {
-            vscode.window.showInformationMessage('Invalid request type')
+        let invalidMessage: string | undefined
+
+        if (!AWS_SAM_DEBUG_REQUEST_TYPES.includes(debugConfiguration.request)) {
+            invalidMessage = localize(
+                'AWS.sam.debugger.invalidRequest',
+                'Debug Configuration has an unsupported request type. Supported types: {0}',
+                AWS_SAM_DEBUG_REQUEST_TYPES.join(', ')
+            )
+        }
+
+        if (!AWS_SAM_DEBUG_TARGET_TYPES.includes(debugConfiguration.invokeTarget.target)) {
+            invalidMessage = localize(
```

Comment:
should it be `else if ...` ? This potentially overwrites `invalidMessage`.

Reply:
Ack, not an `else if`, this was two separate statements that returned on their own. Fixing...